python/ray/autoscaler/aws/node_provider.py

```diff
@@ -40,15 +43,59 @@ def __init__(self, provider_config, cluster_name):
         # Try availability zones round-robin, starting from random offset
         self.subnet_idx = random.randint(0, 100)
 
+        self.tag_cache = {}  # Tags that we believe to actually be on EC2.
+        self.tag_cache_pending = {}  # Tags that we will soon upload.
+        self.tag_cache_lock = threading.Lock()
+        self.tag_cache_update_event = threading.Event()
+        self.tag_cache_kill_event = threading.Event()
+        self.tag_update_thread = threading.Thread(
+            target=self._node_tag_update_loop)
+        self.tag_update_thread.start()  # TODO: monitor this?
```

Comment:
Is this still needed?

Reply:
The TODO? I think it could be monitored but it's not essential.

It's more a nice to have. It's hard to see how it could fail; if we wanted to be really sure we could try/except the whole thing.

Consider that the autoscaler will just end itself after multiple failures and is far more susceptible just due to the amount of stuff it's doing

Reply:
Removed.